contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.2.0--2.3.0.sql

```diff
@@ -289,20 +289,77 @@ BEGIN
 END;
 $$ LANGUAGE plpgsql IMMUTABLE STRICT;
 
+-- The sys.table_types_internal view mimics the logic used in sys.is_table_type function
+create or replace view sys.table_types_internal as
+SELECT pt.typrelid
+    FROM pg_catalog.pg_type pt
+    INNER JOIN pg_catalog.pg_depend dep
+    ON pt.typrelid = dep.objid AND pt.oid = dep.refobjid
+    INNER JOIN pg_catalog.pg_class pc ON pc.oid = dep.objid
+    WHERE 
+    pt.typnamespace in (select schema_id from sys.schemas) 
+    AND (pt.typtype = 'c' AND dep.deptype = 'i'  AND pc.relkind = 'r'
+    AND dep.classid = 'pg_catalog.pg_class'::regclass AND dep.refclassid = 'pg_catalog.pg_type'::regclass)
+;
+GRANT SELECT ON sys.table_types_internal TO PUBLIC;
+
+-- The view sys.type_info_internal corresponds to 
+-- type_info_t type_infos[TOTAL_TYPECODE_COUNT] in contrib/babelfishpg_common/src/typecode.c
+-- any changes there need to be reflected here
+create or replace view sys.type_info_internal AS
+select * from
+(
+ values 
+    ('sql_variant'     , 'sql_variant'),
+    ('datetimeoffset'  , 'datetimeoffset'),
+    ('datetime2'       , 'datetime2'),
+    ('datetime'        , 'datetime'),
+    ('smalldatetime'   , 'smalldatetime'),
+    ('date'            , 'date'),
+    ('time'            , 'time'),
+    ('float8'          , 'float'),
+    ('float4'          , 'real'),
+    ('numeric'         , 'numeric'),
+    ('money'           , 'money'),
+    ('smallmoney'      , 'smallmoney'),
+    ('int8'            , 'bigint'),
+    ('int4'            , 'int'),
+    ('int2'            , 'smallint'),
+    ('tinyint'         , 'tinyint'),
+    ('bit'             , 'bit'),
+    ('nvarchar'        , 'nvarchar'),
+    ('nchar'           , 'nchar'),
+    ('varchar'         , 'varchar'),
+    ('bpchar'          , 'char'),
+    ('varbinary'       , 'varbinary'),
+    ('binary'          , 'binary'),
+    ('uniqueidentifier', 'uniqueidentifier'),
+    ('text'            , 'text'),
+    ('ntext'           , 'ntext'),
+    ('image'           , 'image'),
+    ('xml'             , 'xml'),
+    ('decimal'         , 'decimal'),
+    ('sysname'         , 'sysname'),
+    ('rowversion'      , 'timestamp' ),
+    ('timestamp'       , 'timestamp')
+)  t(pg_type_name,tsql_type_name);
+GRANT SELECT ON sys.type_info_internal TO PUBLIC;
```

Comment:
Is it necessary to grant internal object's permission to public?

Reply:
You're right. It is not necessary. I'll remove the grant in the next revision.

Reply:
Hmm, removing the grant complicates testing, because in tests
now I'm getting 
ERROR (Message: permission denied for view type_info_internal)

Is there a way to grant and then revoke a permission for a sys.xyz_internal object in tests?

Reply:
Ok got tests working now after removing the unnecessary grant.